src/CombFilter/CombFilter.cpp

```diff
@@ -0,0 +1,145 @@
+//
+//  CombFilter.cpp
+//  CombFilter
+//
+//  Created by Vedant Kalbag on 09/02/22.
+//
+#include "ErrorDef.h"
+#include "CombFilter.h"
+
+
+// =====================================================================================================
+// CCombFilterBase method definitions
+CCombFilterBase::CCombFilterBase(int delayLength, int iNumChannels, float gainValue)
+{
+    m_iNumChannels = iNumChannels;
+    m_fGainValue = gainValue;
+    m_iDelayValueSamples = delayLength;
+    m_RingBuffer = new CRingBuffer<float>* [m_iNumChannels];
+    for (int i=0;i<m_iNumChannels;i++)
+    {
+        m_RingBuffer[i] = new CRingBuffer<float>(m_iDelayValueSamples);
```

Comment:
For every `new`, a `delete`; this needs a destructor. (Better to have it here than in the subclasses, since this is where the allocation happens.)